Assignment6.java

```diff
@@ -0,0 +1,75 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class Assignment6 {
+    /**
+     * Rearranges given to make it the same as desired.
+     *
+     * @param given   initial version of the array of numbers.
+     * @param desired final version of the array of numbers.
+     * @return list of moves made during arrangement.
+     */
+    public static List<List<Integer>> rearrange(int[] given, int[] desired) {
+        // Keys are values in given, values are their indexes in given.
+        Map<Integer, Integer> indexMap = new HashMap<>();
+        // Generated for testing purposes.
+        List<List<Integer>> listOfMoves = new LinkedList<>();
+
+        for (int i = 0; i < given.length; i++) {
+            indexMap.put(given[i], i);
+        }
+
+        for (int i = 0; i < desired.length; i++) {
+            // Already in its desired place.
+            if (indexMap.get(desired[i]) == i) {
+                continue;
+            }
+
+            int indexOf0 = indexMap.get(0);
+
+            // If i isn't already empty.
+            if (indexOf0 != i) {
+                // Move the number in i to the empty slot so that i becomes empty.
+                indexMap.put(0, i);
+                indexMap.put(given[i], indexOf0);
+
+                given[indexOf0] = given[i];
+                given[i] = 0;
+
+                printMove(i, indexOf0);
```

Comment:
The `printMove` method's only task is to print the move. I didn't really understand what you wanted to say.